### PR TITLE
Mttl 3453

### PR DESCRIPTION
**Pull request by oggg**
## Link to JIRA ticket
[Mttl 3453](https://hackney.atlassian.net/browse/MTTL-3453)
## Describe this PR
Implement sorting for housing search processes

### CODE REVIEW

File: HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        private SortFactory _sut;
+
+        public ProcessListSortFactoryTests()
+        {
+            _sut = new SortFactory();
+        }
+
+        [Fact]
+        public void ShouldNotSortAsDefault()
```
Comment (notatiyyah) [inline HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs:19]: ShouldSortAsDefault?

File: HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            // Arrange + Act
+            var result = _sut.Create<QueryableProcess, GetProcessListRequest>(new GetProcessListRequest());
```
Comment (notatiyyah) [inline HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs:22]: Can you add a test or another case for if SortBy is an empty string?

File: HousingSearchApi/V1/Infrastructure/Sorting/RelatedEntityTypeAsc.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+using Hackney.Shared.HousingSearch.Gateways.Models.Processes;
+using HousingSearchApi.V1.Interfaces.Sorting;
+using Nest;
+using System.Linq;
+
+namespace HousingSearchApi.V1.Infrastructure.Sorting
+{
+    public class RelatedEntityTypeAsc : ISort<QueryableProcess>
+    {
```
Comment (notatiyyah) [inline HousingSearchApi/V1/Infrastructure/Sorting/RelatedEntityTypeAsc.cs:9]: I think renaming to PersonNameAsc/Desc will be clearer - especially if we add sorting by address down the line

File: HousingSearchApi.Tests/V1/E2ETests/Stories/GetProcessesStories.cs [hunk 4]
```diff
@@ -112,5 +112,19 @@ public void ServiceReturnsCorrectPageSize()
                 .Then(t => _steps.ThenTheReturningResultsShouldBeOfThatSize(3))
                 .BDDfy();
         }
+
+        [Theory]
+        [InlineData("name")]
+        [InlineData("process")]
+        [InlineData("patch")]
+        [InlineData("state")]
+        public void ServiceReturnsCorrectSort(string sortBy)
```
Comment (notatiyyah) [inline HousingSearchApi.Tests/V1/E2ETests/Stories/GetProcessesStories.cs:121]: Can you do something similar for descending?

File: HousingSearchApi.Tests/V1/E2ETests/Fixtures/BaseFixture.cs [hunk 5]
```diff
@@ -58,5 +62,37 @@ protected void EnsureEnvVarConfigured(string name, string defaultValue)
                 _elasticSearchAddress = default;
             }
         }
+
+        protected static List<List<RelatedEntity>> GenerateRelatedEntities(int count)
```
Comment (notatiyyah) [inline HousingSearchApi.Tests/V1/E2ETests/Fixtures/BaseFixture.cs:66]: I think this should go into the process fixture as it's specifically for processes, not any other data entity

File: HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs (hunk 2 above)
Comment (oggg) [reply HousingSearchApi.Tests/V1/Helper/ProcessListSortFactoryTests.cs:22]: it will hit the DefaultSort as there will be no matching with the sortBy values and I think this test is covered by the first case from that file

**Review by notatiyyah — APPROVED** [review]
